analyzers/tests/SonarAnalyzer.Test/TestCases/IfCollapsible.cs

```diff
@@ -65,4 +66,29 @@ public void Test(bool cond1, bool cond2, bool cond3)
             }
         }
     }
+
+    // Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/9221
+    public class Repro_9221
+    {
+        public DateTime? ThisWorks()
+        {
+            dynamic anything = "2024-04-29";
+
+            if (anything is string) // Secondary
+                if (DateTime.TryParseExact(anything, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dt)) // Noncompliant FP
+                    return dt;
+
+            return null;
```

Comment:
I would add more test cases based on what @pavel-mikula-sonarsource said about not raising if there is a dynamic in the statement:
```cs
if (notDynamic is not null)
  if (notDynamic == "something" && anything is string) // Noncompliant (not sure if FP or not there)
    return null;
    
if (notDynamic is not null)
  if (notDynamic == "something") // Noncompliant (to make sure we do not deactivate S1066 for the whole block)
    return null;
```